src/WidgetApi/WidgetApi.js

```diff
@@ -0,0 +1,23 @@
+import ApiConnectionIframe from "./ApiConnectionIframe";
+import ApiEventsRegistry from "./ApiEventsRegistry";
+
+class WidgetApi {
+
+    constructor() {
+        if (this.detectIframe()) {
+            let apiConnection = new ApiConnectionIframe()
+            new ApiEventsRegistry(apiConnection)
+            apiConnection.start()
+        }
+    }
+
+    detectIframe () {
+        try {
+            return window.self !== window.top;
```

Comment:
Can this actually throw?

Reply:
This is the source of this function: [https://stackoverflow.com/questions/326069/how-to-identify-if-a-webpage-is-being-loaded-inside-an-iframe-or-directly-into-t](https://stackoverflow.com/questions/326069/how-to-identify-if-a-webpage-is-being-loaded-inside-an-iframe-or-directly-into-t)